README.md
<!-- This page can be read using your web browser -->
<!-- Written by Travis Moore on 9-23-15 -->
<h1>README.md</h1>
<h2>Package: erp.easy</h2>
<hr />
<h3>What is erp.easy?</h3>
<p>So you&#8217;ve recorded and cleaned (processed) some ERP data... Now what? If you&#8217;re not a programmer,
or are new to ERPs, the next step may be a bit daunting, or at the very least, time consuming
if done by hand. The erp.easy package provides an intuitive approach to exploring your processed
data, without requiring a background in computer programming. The erp.easy package provides
three categories of functions, optimized to be easy to use: loading ERP files, plotting ERP data, and
analyzing ERP data.</p>
<br />
<h3>What can I do with erp.easy?</h3>
<p><b>Loading Functions</b></p>
<p>The function load.data exists to save you the hassle of opening each individual ERP file and
adding a header or other identifying information to the files. This function expects data formatted
with electrodes across the columns and time points as rows. Additional columns (i.e., "Subject",
"Stimulus", and "Time") will be added upon import to help organize your data. The erp.easy package
will use existing headers provided in raw data files to refer to electrodes for use in functions,
or will assign headers if none are present (see load.data.) Single electrodes can be passed to the
package functions, or several electrodes can be provided (i.e., when using dense arrays) and those
electrodes will be averaged together as a single electrode. Also see the function easy.load for a 
user-friendly way to import your data using the erp.easy data import wizard.</p>
<p><b>See help files/pdf manual for examples</b></p>
<br />
<h3>Plotting Functions</h3>
<p>The plotting functions grandaverage, individual, and mosaic provide several ways to visualize
both grand average and individual data. Color-coding and labeling happens automatically for ease
of use.</p>
<p>Example: <b>grandaverage(ERPdata, electrodes = "V78")</b></p>
<br />
<h3>Analysis Functions</h3>
<p>The analysis functions m.measures and p.measures calculate standard ERP measures such as
mean amplitude, standard deviation, peak amplitude and peak latency for both grand average and
individual waveforms.</p>
<p>Example: <b>m.measures(ERPdata, electrodes = "V78", window = c(1000, 1500))</b></p>



NEWS.md
---
# erp.easy 1.1.0
### March 1st, 2017
---

## New Functions
### easy.load:
* A more user-friendly way to import your data. Just
    answer a few questions and the function generates 
	a ready to copy/paste line of code.

## Updates and Fixes to Previously Existing Functions
### load.data:
* Option to perform baseline correction on data
	as it is loaded
* Labels data using specified subject values instead 
    of generic numbers

### grandaverage:
* Option to suppress the legend (for cases where it covers
    part of the waveform)
* Indicates the electrodes used to create the plot

### p.measures:
* Option to search for the most negative, positive
    or absolute peak value. Previously only absolute value was
	allowed (i.e., greatest deviation from 0 regardless of polarity).
	Option to use simple or local peak measures.
* Existing scripts using p.measures will have to be updated by 
    adding the "pol" argument with an acceptable parameter

### individual:
* Returns the individual average data for each subject
	in each condition
* Option to create separate plots for each individual by 
    setting plots="y". By default, no plots are returned (i.e., 
    only raw data values are returned)

### erp.filter
* This function has been removed.


---
# erp.easy 1.0.0
### January 29th, 2016
---

## New Functions
### erp.filter:
* Uses the "signal" package to allow Butterworth  
	filterting of erp.easy data objects

## Updates and Fixes to Previously Existing Functions
### all plotting functions:
* There is now a solid background for all plot legends

### dif.wave:
* Can now specify Stimulus name for difference wave
* More descriptive default for Stimulus name for
	difference wave, if none is specified
* Can now specify which data frame to keep with the
	difference wave (if any)

### grandaverage:
* Raw amplitude data for the grand average is returned
    for each condition

### load.data:
* Change in acceptable file name format means files no longer 
	need to be numbered.  Now all text preceding an "_" will be
	ignored, and all text following an "_" will be read as
	the specified condition.  This allows removing or adding
	subject files without having to renumber existing files.  

### m.measures: 
* Mean erp measures are automatically returned with a plot 
	of the grand average and time window

### p.measures:
* Peak erp measures are automatically returned with: 
	+ a plot of the grand average
	+ the time window
	+ a vertical line indicating the peak latency
* Fixed a bug that produced inaccurate local peak measures for 
	individual data (grand average peak data were accurate)

	
---
# erp.easy 0.6.4
### September, 2015
---
## Initial Release

R/erp.easy.R
#' erp.easy: A user-friendly package for exploring event-related potential (ERP) data
#'
#' So you've recorded and cleaned (processed) some ERP data... Now what?
#' If you're not a programmer, or are new to ERPs, the next step may be a bit daunting, or
#' at the very least, time consuming if done by hand. The erp.easy package provides an
#' intuitive approach to exploring your processed data, without requiring a background
#' in computer programming. The erp.easy package provides three categories of functions,
#' optimized to be easy to use: loading ERP files, plotting ERP data, and analyzing ERP data.
#'
#' @section Loading functions:
#' The function \code{\link{load.data}} exists to save you the hassle of opening each individual
#' ERP file and adding a header or other identifying information to the files.  This function
#' expects data formatted with electrodes across the columns and time points as rows. Additional
#' columns (i.e., "Subject", "Stimulus", and "Time") will be added upon import to help organize your
#' data. The erp.easy package will use existing headers provided in raw data files to refer to electrodes
#' for use in functions, or will assign headers if none are present (see \code{\link{load.data}}.)
#' Single electrodes can be passed to the package functions, or several electrodes can be provided
#' (i.e., when using dense arrays) and those electrodes will be averaged together as a single electrode.
#' See also \code{\link{easy.load}} for a more user-friendly way to import your data using the erp.easy
#' data import wizard.
#'
#' @section Plotting functions:
#' The plotting functions \code{\link{grandaverage}}, \code{\link{individual}}, and
#' \code{\link{mosaic}} provide several ways to visualize both grand average and
#' individual data. Color-coding and labeling happens automatically for ease of use.
#'
#' @section Analysis functions:
#' The analysis functions \code{\link{m.measures}} and \code{\link{p.measures}} calculate
#' standard ERP measures such as mean amplitude, standard deviation, peak amplitude and
#' peak latency for both grand average and individual waveforms.
#'
#' @author Travis Moore
#'
#' @examples
#' library(erp.easy)
#'
#' data(ERPdata)
#'
#' grandaverage(ERPdata, electrodes = "V78")
#'
#' mosaic(ERPdata, electrodes = "V78")
#'
#' m.measures(ERPdata, electrodes = "V78", window = c(1000, 1500))
#'
#' @docType package
#' @name erp.easy
NULL
#> NULL


